tests/integration_tests/test_examples.py

```diff
@@ -1,39 +1,84 @@
 """
 This is an integration test to make sure scripts from examples/
-work when running `python examples/xx/xxx.py`.
+work when executing `./examples/[*/]*.py`.
 """
-from garage.experiment import LocalRunner, run_experiment
-from garage.np.baselines import LinearFeatureBaseline
-from garage.tf.algos import VPG
+import os
+import subprocess
+import tempfile
+from unittest.mock import Mock
+
+import pytest
+
+from garage.envs import GridWorldEnv
+from garage.np.policies import ScriptedPolicy
 from garage.tf.envs import TfEnv
-from garage.tf.policies import CategoricalMLPPolicy
 
+examples_root_dir = 'examples/'
+blacklist = [
+    os.path.join(examples_root_dir, 'resume_training.py'),
+]
+sim_policy_path = os.path.join(examples_root_dir, 'sim_policy.py')
+
+
+def enumerate_examples():
+    examples = []
+    for (dirpath, _, filenames) in os.walk(examples_root_dir):
+        for file in filenames:
+            filepath = os.path.join(dirpath, file)
+            if filepath not in blacklist and file.endswith(
+                    '.py') and file != sim_policy_path:
+                examples.append(filepath)
+    return examples
 
-def _run_task(snapshot_config, *_):
-    with LocalRunner(snapshot_config=snapshot_config) as runner:
-        env = TfEnv(env_name='CartPole-v1')
 
-        policy = CategoricalMLPPolicy(name='policy',
-                                      env_spec=env.spec,
-                                      hidden_sizes=(32, 32))
+@pytest.mark.parametrize('filepath', enumerate_examples())
+def test_examples(filepath):
+    args = [
+        filepath,
+    ]
+    if filepath in {
+            os.path.join(examples_root_dir, 'step_env.py'),
```

Comment:
this is kind of complicated. why not just make a different test method for these?